tests/common/auth_test.go

```diff
@@ -0,0 +1,1303 @@
+			t.Fatalf("want %v error but got %v error", rpctypes.ErrInvalidAuthMgmt, err)
+		}
+		_, err = rootAuthClient.UserRevokeRole(ctx, "root", "root")
+		if err == nil || !strings.Contains(err.Error(), rpctypes.ErrInvalidAuthMgmt.Error()) {
+			t.Fatalf("want %v error but got %v error", rpctypes.ErrInvalidAuthMgmt, err)
+		}
+	})
+}
+
+func TestAuthLeaseTestKeepAlive(t *testing.T) {
+	testRunner.BeforeTest(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	clus := testRunner.NewCluster(ctx, t, config.WithClusterConfig(config.ClusterConfig{ClusterSize: 1}))
+	defer clus.Close()
+	cc := testutils.MustClient(clus.Client())
+	testutils.ExecuteUntil(ctx, t, func() {
+		authEnable(ctx, cc, t)
+		rootAuthClient := testutils.MustClient(clus.Client(WithAuth("root", "root")))
+		authSetupDefaultTestUser(ctx, rootAuthClient, t)
+		resp, err := rootAuthClient.Grant(ctx, 10)
+		if err != nil {
+			t.Fatal(err)
+		}
+		leaseID := resp.ID
+		if err = rootAuthClient.Put(ctx, "key", "value", config.PutOptions{LeaseID: leaseID}); err != nil {
+			t.Fatal(err)
+		}
+		if _, err = rootAuthClient.KeepAliveOnce(ctx, leaseID); err != nil {
+			t.Fatal(err)
+		}
+		gresp, err := rootAuthClient.Get(ctx, "key", config.GetOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(gresp.Kvs) != 1 || string(gresp.Kvs[0].Key) != "key" || string(gresp.Kvs[0].Value) != "value" {
+			t.Fatalf("want kv pair ('key', 'value') but got %v", gresp.Kvs)
+		}
+	})
+}
+
+func TestAuthLeaseTestTimeToLiveExpired(t *testing.T) {
+	tcs := []struct {
+		name       string
+		JWTEnabled bool
+	}{
+		{
+			name:       "JWTEnabled",
+			JWTEnabled: true,
+		},
+		{
+			name:       "JWTDisabled",
+			JWTEnabled: false,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			testRunner.BeforeTest(t)
+			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+			defer cancel()
+			clus := testRunner.NewCluster(ctx, t, config.WithClusterConfig(config.ClusterConfig{ClusterSize: 1}))
+			defer clus.Close()
+			cc := testutils.MustClient(clus.Client())
+			testutils.ExecuteUntil(ctx, t, func() {
+				authEnable(ctx, cc, t)
+				rootAuthClient := testutils.MustClient(clus.Client(WithAuth("root", "root")))
+				authSetupDefaultTestUser(ctx, rootAuthClient, t)
+				resp, err := rootAuthClient.Grant(ctx, 2)
+				if err != nil {
+					t.Fatal(err)
+				}
+				leaseID := resp.ID
+				if err = rootAuthClient.Put(ctx, "key", "val", config.PutOptions{LeaseID: leaseID}); err != nil {
+					t.Fatal(err)
+				}
+				// eliminate false positive
+				time.Sleep(3 * time.Second)
+				tresp, err := rootAuthClient.TimeToLive(ctx, leaseID, config.LeaseOption{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				if tresp.TTL != -1 {
+					t.Fatalf("want leaseID %v expired but not", leaseID)
+				}
+				gresp, err := rootAuthClient.Get(ctx, "key", config.GetOptions{})
+				if err != nil || len(gresp.Kvs) != 0 {
```

Comment:
Note, no need to fix it in this PR. This checking for empty response will really benefit from rewrite to testify.